starlite/contrib/jwt/middleware.py

```diff
@@ -0,0 +1,164 @@
+from typing import TYPE_CHECKING, Awaitable, List, Optional, Union
+
+from starlite import (
+    AbstractAuthenticationMiddleware,
+    AuthenticationResult,
+    NotAuthorizedException,
+)
+from starlite.connection import ASGIConnection
+from starlite.contrib.jwt.jwt_token import Token
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from starlite.types import ASGIApp, Scopes
+    from starlite.utils import AsyncCallable
+
+
+class JWTAuthenticationMiddleware(AbstractAuthenticationMiddleware):
+    """JWT Authentication middleware.
+
+    This class provides JWT authentication functionalities.
+    """
+
+    def __init__(
+        self,
+        app: "ASGIApp",
+        algorithm: str,
+        auth_header: str,
+        exclude: Optional[Union[str, List[str]]],
+        exclude_opt_key: str,
+        retrieve_user_handler: "AsyncCallable[[Token, ASGIConnection[Any, Any, Any]], Awaitable[Any]]",
+        scopes: "Scopes",
+        token_secret: str,
+    ):
+        """Check incoming requests for an encoded token in the auth header specified, and if present retrieves the user
```

Comment:
```suggestion
        """Check incoming requests for an encoded token in the auth header specified, and if present retrieve the user
```

This should go into the class docstring.